### ISSUE TITLE
Submarine editor restrictions set for light components limit workflow and creativity. 

### ISSUE THREAD

**Issue by fr45** [Regalis11/Barotrauma#8867]
- [ V] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Recently introduced restrictions for amount of lights and shadow-casting lights needlessly restrict workflow and user creativity.
My default style of making submarines is to paste a few of them in a single file, so that I don't have to switch between different subs when I want to look closer at this or that component or at this or that hull section part.
Now I can't do it.
My rig can handle it, game can load that file, why there needs to be such restriction for the editor - I do not understand.
Please, revert this change as soon as possible.

**Steps To Reproduce**
Open a sub file in editor, paste a few other sub files in it, try to save.

**Version**
Urban Expanses update, 20 april 2022. Windows x64.

**Additional information**
![image](https://user-images.githubusercontent.com/1794500/164309147-1c00b6ca-1142-4988-b35f-c8d4bf45a4ad.png)

**Comment by SeaverBeaver** [comment]
I'd have to agree, being forced to optimize as you are building isn't a great solution to try to improve performance, and hampers the design process. A lot of submarine builders build what is in their head first, and optimize once they have it down, which is no longer possible.

I'd recommend preventing publishing a submarine to the workshop if it doesn't meet the requirements, rather than preventing saving. And considering many _heavily_ modded "maps" made in the sub editor are the size of small continents, having a warning  attached to the sub that it doesn't meet the requirements could also be a solution, albeit a niche one.

**Comment by Regalis11** [comment]
I agree that we may have been a little too heavy-handed with this, but it seemed that the warnings when saving just weren't enough, and that many sub builders ignored them entirely and then flooded us with complaints and issue reports when a submarine with an excessive number of lights didn't run properly.

There is currently a hidden way to bypass the restriction though: you need to enter "debugdraw" in the debug console before saving.

**Comment by SeaverBeaver** [comment]
Fair enough, and thanks for the tip. I can certainly see where your coming from…

“I’ve put 300 shadow casting lights in my sub and my performance is affected pls fix”

**Comment by Rokvach** [comment]
As I suggested some time ago, I think we shouldn't restrict saving the submarines but instead restrict placing new lights.

So something like this
1. Player starts placing lights
2. Player hits the maximum shadow casting light limit
3. Now newly placed lights will automatically have the shadow casting disabled
4. Player hits the maximum allowed lights
5. No new lights can't be placed, trying to place a light shows a marquee error that the maximum amount has been reached
6. Now player can adjust the existing lights to delete ones they don't necessarily need and enable shadow casting on the lights which actually will need it (by disabling it on the ones that doesn't need it). 

This way the submarine file can always be saved even if it does go above the maximum amount of lights and the submarines saved previously that does exceed the maximum amount of lights still has the the previous amount of lights but they just can't add any new ones.

**Comment by Tsunder** [comment]
> 1. Player starts placing lights
> 2. Player hits the maximum shadow casting light limit
> 3. Now newly placed lights will automatically have the shadow casting disabled
> 4. Player hits the maximum allowed lights
> 5. No new lights can be placed, trying to place a light shows a marquee error that the maximum amount has been reached
> 6. Now player can adjust the existing lights to delete ones they don't necessarily need and enable shadow casting on the lights which actually will need it (by disabling it on the ones that doesn't need it).

this seems even more heavy handed (and player hostile) since it actively constantly interferes with sub editing/building

**Comment by Regalis11** [comment]
> this seems even more heavy handed (and player hostile) since it actively constantly interferes with sub editing/building

I agree with this, also I think another problem with that approach would be that it's easy to accidentally (or intentionally) go above the soft limit, which would lead to non-shadow-casting lights looking weird in the sub. Better to force the sub builder to address the problem than trying to address it automatically by disabling shadow casting, which would most likely lead to non-optimal results. 

> I think a warning every save (like the warnings with unlinked vents/unattached gaps) would suffice.

I don't think it suffices, we used to have that and everyone just ignored it.
